### PR TITLE
Re-former Project Review Request

### PR DESCRIPTION
**Pull request by rloterh**
- All requirements for this Rails form project are met:
- Created form for User model with form helper 'form_for'
- Form can be validated and throw appropriate error message 
- Form saves validated input correctly

### CODE REVIEW

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

Hello!
Great job so far with this project! 👍 
Well done setting up the linters! 💯 
Nice and clean documentation and commits! 👏 
Clean and readable code! 🥇 
However, we need to implement some changes in order to get this project approved 💪 
Kindly check my comments below 👇  and resubmit your project once the changes have been done ⏩ 
## Mandatory changes
- In your views, you should have commented out the previous forms you have built. In this project, we need to be sure you can follow all the steps building different form, from the most basic to the most complex. Kindly, add your previous forms( form and form_tag) and comment them out.
- In your users controller, in the create index, you can use your private method to pass the params, actually, you have this method for that purpose. ie. @user = User.new(user_params), it makes your code cleaner and more readable.
- Make your root page http://localhost:3000/users/new, it will make easier the user to use your app, it is not necessary to land on the rails welcome page
- Create a show action to see the user information, and add an update/edit button that takes you to the edit form to modify the data. Once the user is logged in, it should redirect to its show page and from there to the edit page
- Add a message or alert when the user is updated and an error message when the update was not successful
## [OPTIONAL]
- Remember you can use a partial to create a form and use it wherever you want without repeating code
- In your readme file, on the author section, it would be better if you change the links showing your name. ie.Github: @rloterh

**Comment by rloterh** [comment]
Please, have fixed issues as recommended by first TSE
added comment code, refactor the new method in the controller and fixed the welcome page

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users, only: %i[ index new create edit update show ]
+  root 'user#new'
```
Comment (lekegitrepo) [inline config/routes.rb]: - **_This error showed up when I tried to run the project_**

Action-Controller-Exception-caught

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
#### Hi @rloterh :+1: you have done a good job so far. However, there is an error when I tried to run the project. Check the image below.

#### _Reviewed by Temitayo Adeleke_

**Comment by rloterh** [comment]
Resolved issue as per second review
- Application is loading correctly

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <%= label_tag :email %>
+  <%= text_field_tag :email, params[:email] %>
+  <%= label_tag :password %>
+  <%= password_field_tag :password, params[:password] %>
+  <%= submit_tag "Submit" %>
+<% end %>
+<% 
+=end %>
+
+<%# form_for %>
```
Comment (krys2fa) [inline app/views/users/new.html.erb]: - [ ] [MANDATORY] Your form_for implementation is commented out as requested but it still needs to be completed as below:
```
<%#= form_for @user do |u|%> 
<%#= u.label :username %> <%= u.text_field :username%> 
...
...
<%#= u.submit %> 
<%# end %>
```
NB: Remember to fill out all the fields.

**Review by krys2fa — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
Well done on getting your application to load @rloterh. 👏 
You have also implemented most of the earlier requested changes. Here are a few you missed:

- [ ] [MANDATORY] Create a show action to see the user information, and add an update/edit button that takes you to the edit form to modify the data. Once the user is logged in, it should redirect to its show page and from there to the edit page.

- [ ] [MANDATORY] Add a message or alert when the user is updated and an error message when the update was not successful.

- [ ] [OPTIONAL] In your readme file, on the author section, it would be better if you change the links showing your name. ie.Github: @rloterh

Kindly request another review after making the changes.

Happy Coding,
Chris.

**Comment by rloterh** [comment]
Resolved all issues as per third review

- Application is functioning correctly

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED ♻️ 

Good job with the implemented changes 👍 
Your project is approved 🥳 
Remember to fill up the completion form 📑  and you can move into the next project ⏩ 
Good luck 😺

**Pull request merged**